igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/TravisCache.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2016 Schibsted ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.travis
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Service
+import redis.clients.jedis.Jedis
+import redis.clients.jedis.JedisPool
+
+
+/*
+    This creates an internal queue for igor triggered travis jobs.
+    The travis api does not return an id we can track in the queue.
+ */
+@Service
+@ConditionalOnProperty("travis.enabled")
+class TravisCache {
+    @Autowired
+    JedisPool jedisPool
+
+    @SuppressWarnings('GStringExpressionWithinString')
+    @Value('${spinnaker.jedis.prefix:igor}')
+    String prefix
+
+    String id = 'builds:queue'
+
+    Map getQueuedJob(String master, int queueNumber) {
+        Jedis resource = jedisPool.resource
+        if (!resource.exists(makeKey(master, queueNumber))) {
```

Comment:
Thanks @lwander, I solved this a little bit different in https://github.com/spinnaker/igor/pull/80/commits/1da9404c3b72b297bd8328b6139e38d2b5225728 . I surrounded the calls with Jedis.withCloseable. That should save if redis fails.